mempool/mempool.go

```diff
@@ -22,6 +22,14 @@ import (
 	"github.com/tendermint/tendermint/types"
 )
 
+// PreCheckFunc is an optional filter to determine if a transaction should be
+// rejected. Invoked before CheckTx.
+type PreCheckFunc func(types.Tx) bool
```

Comment:
Perhaps add that an example usage would be to ensure a tx isn't too large to ever fit into a block?